docs/draft/notifications.md

```diff
@@ -0,0 +1,49 @@
+
+## Local Notifications
+
+Using local notifications, we can offer notifications for any event in the app without any privacy cost and dependency on third party services.
+
+### Android
+
+Android allows running services on the device. When the user enables notifications, the client starts a `Foreground Service`, and a permanent notification indicating that the service is running is displayed to the user, as required by Android guidelines.
+The service simply keeps the app from being killed by the system when it is in the background.
+
+In the Android client, events from `status-go` are handled by the `handleSignal` method of `im.status.ethereum.module`. Those signals are passed to the clojurescript part of the client. This is the general behavior when the client is in the foreground. For the local notifications, we also react on these signals when the app is in the background. Currently, we only display a notification for 1-1 messages, but support for any signal that status-go may send can be easily added.
+
+The signals are passed to `NewMessageSignalHandler` which takes care of managing the local notifications. It keeps track of the current active notifications so that new messages are displayed together by chat as it is done in most common chat applications.
+
+Notifications are a separate process in Android, and interaction with a notification generates an `Intent`. To handle intents, the `NewMessageSignalHandler` uses a `BroadcastReceiver`, in order to update the state of local notifications when the user dismisses or tap a notification. If the user taps on a notification, the `BroadcastReceiver` generates a new intent to open the app and uses universal links to get the user to the right place.
+
+
+### iOS
+
+We are not able to offer local notifications on iOS because there is no concept of services in iOS. It offers background updates but they’re not consistently triggered, and cannot be relied upon. The system decides when the background updates are triggered and the heuristics aren't known.
+
+## Future development
+
+### More types of local notifications
+
+We can and should implement more types of local notifications on Android so that users can be notified for instance when:
+- they are mentioned in any chat, 
+- a keyword they are following appears in any chat, 
+- they receive a transaction or a transaction they sent has been confirmed,
+- the total value of their assets in the wallet reached a threshold
+
+### Notifications for iOS
+
+Since iOS cannot reliably display local notifications, we need to find an alternative. But we should not impact Android users with iOS limitations. 
```

Comment:
## Copyright

Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).